src/main/java/com/mitchej123/hodgepodge/mixins/fixHungerOverhaul/MixinHungerOverhaulHealthRegen.java

```diff
@@ -0,0 +1,47 @@
+package com.mitchej123.hodgepodge.mixins.fixHungerOverhaul;
+
+import cpw.mods.fml.common.eventhandler.SubscribeEvent;
+import iguanaman.hungeroverhaul.HungerOverhaul;
+import iguanaman.hungeroverhaul.config.IguanaConfig;
+import iguanaman.hungeroverhaul.food.FoodEventHandler;
+import squeek.applecore.api.hunger.HealthRegenEvent;
+import net.minecraft.world.EnumDifficulty;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+
+@Mixin(value = FoodEventHandler.class, remap = false)
+public class MixinHungerOverhaulHealthRegen {
+
+    /**
+     * @author mitchej123
+     * @reason Fix bad interaction with SoL Carrot Edition
+     */
+    @Overwrite()
+    @SubscribeEvent
+    public void onHealthRegenTick(HealthRegenEvent.GetRegenTickPeriod event) {
+        float wellfedModifier = 1.0F;
+        if (event.player.isPotionActive(HungerOverhaul.potionWellFed))
+            wellfedModifier = 0.75F;
+
+        EnumDifficulty difficulty = event.player.worldObj.difficultySetting;
+        float difficultyModifierHealing = 1.0F;
+        if (IguanaConfig.difficultyScalingHealing) {
+            if (difficulty.getDifficultyId() <= EnumDifficulty.EASY.getDifficultyId())
+                difficultyModifierHealing = 0.75F;
+            else if (difficulty == EnumDifficulty.HARD)
+                difficultyModifierHealing = 1.5F;
+        }
+
+        float lowHealthModifier = 1.0F;
+        if (IguanaConfig.modifyRegenRateOnLowHealth) {
```

Comment:
This formula needs 1) to be tested and 2) be adjusted -- right now I think it'll heal _too_fast when hearts are over 20.

That being said, this mixin works as is, just needs tweaks